addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -569,6 +588,7 @@ def _get_future_leaves_on(self, accrual_date):
                 and (not self.nextcall or self.nextcall <= accrual_date)):
             return 0
 
+        self._invalidate_cache()
```

Comment:
Why is it needed?

Reply:
As i explained in the commit message, if we create a new record it first browse the cache looking if the NewId is already existant. If it is, we get the value from cache, meaning if we did some treatment on  them it persist for the second call. So calling the function 2 consecutive time with same parameter can have different output.